public/app/core/components/code_editor/code_editor.ts

```diff
@@ -143,7 +143,7 @@ function link(scope, elem, attrs) {
   });
 
   function setLangMode(lang) {
-    ace.acequire('ace/ext/language_tools');
+    // ace.acequire('ace/ext/language_tools'); // TODO: Do we need this?
```

Comment:
$1M question

Reply:
so I think it has to do with basic autocomplete, https://github.com/ajaxorg/ace/wiki/How-to-enable-Autocomplete-in-the-Ace-editor 

But I test this branch and the autocomplete seemed to work, only one thing did not work and that was prometheus custom auto complete for time ranges when you hit `[` key you should get a range vector suggestions. 

but we might be able to ignore that as we soon will us the slate editor for Prometheus. 
https://github.com/grafana/grafana/pull/15364 

I tested the mysql & posgres data sources and the auto complete and text highlight looked ok, did not test the json editors nor the text panel yet

Reply:
After some debugging:
The prometheus custom auto complete is not working anymore since in older versions of ace, the logic matching input with autocomplete results preferred matching against `item.value`. 

`var caption = item.value || item.caption || item.snippet;`

But in latest ace it's changed to item.caption:
`var caption = item.caption || item.value || item.snippet;`

Since we append the `[` to the item.value, we will always get zero results

-----
For reference: An item object can look like this:

```
{
   caption: "1s"
   meta: "range vector"
   value: "[1s"
}
```